File: Runners/JavaCodeRunner/src/main/java/csc/coderunner/CompilationResult.java
package csc.coderunner;

import java.util.List;

public class CompilationResult
{
	private boolean success;
	private List<CompileError> errors;

	public CompilationResult(boolean success, List<CompileError> errors)
	{
		this.success = success;
		
		if (!success)
			this.errors = errors;
	}
	
	/**
	 * @return Whether or not the compilation was successful.
	 */
	public boolean getSuccess()
	{
		return success;
	}

	/**
	 * @return The compiler errors, if the compilation was not successful.
	 */
	public List<CompileError> getErrors()
	{
		return errors;
	}
}

File: Services/src/BuildService/BuildService.Model/CodeRunner/CompilationResult.cs
using System.Collections.Generic;

namespace CSC.BuildService.Model.CodeRunner
{
	/// <summary>
	/// The result of a compilation.
	/// </summary>
	public class CompilationResult
	{
		/// <summary>
		/// Whether or not the compilation succeeded.
		/// </summary>
		public bool Success { get; set; }

		/// <summary>
		/// The compile errors, if the compilation failed.
		/// </summary>
		public List<CompileError> Errors { get; set; }
	}
}